cc_types.c

```diff
@@ -290,6 +292,69 @@ void create_struct(void)
 	i->members = last;
 }
 
+void create_enum(void)
+{
+	struct type* head = calloc(1, sizeof(struct type));
+	require(NULL != head, "Exhausted memory while creating an enum\n");
+
+	head->name = global_token->s;
+	head->type = head;
+	head->indirect = NULL;
+	head->next = global_types;
+
+	head->size = register_size; /* We treat enums as always being ints. */
+	head->is_signed = TRUE;
+
+	global_types = head;
+
+	global_token = global_token->next;
+	require(NULL != global_token, "Incomplete enum definition at end of file\n");
```

Comment:
sorry, I meant this has to go after `require_match`

Reply:
Wouldn't that then never be run since `require_match` exits if `NULL == global_token`?

```c
void require_match(char* message, char* required)
{
	if(NULL == global_token)
	{
		line_error();
		fputs("EOF reached inside of require match\n", stderr);
		fputs("problem at end of file\n", stderr);
		fputs(message, stderr);
		exit(EXIT_FAILURE);
	}
	if(!match(global_token->s, required))
	{
		line_error();
		fputs(message, stderr);
		exit(EXIT_FAILURE);
	}
	global_token = global_token->next;
}
```

Reply:
Nvm, I understand now.

Reply:
Moved to after the `require_match`.